streams/examples/src/main/java/org/apache/kafka/streams/examples/temperature/TemperatureDemo.java

```diff
@@ -0,0 +1,109 @@
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KStreamBuilder;
+import org.apache.kafka.streams.kstream.KeyValueMapper;
+import org.apache.kafka.streams.kstream.Predicate;
+import org.apache.kafka.streams.kstream.Reducer;
+import org.apache.kafka.streams.kstream.TimeWindows;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.kstream.internals.WindowedDeserializer;
+import org.apache.kafka.streams.kstream.internals.WindowedSerializer;
+
+import java.util.Properties;
+
+/**
+ * Demonstrates, using the high-level KStream DSL, how to implement an IoT demo application
+ * which ingests temperature value processing the maximum value in the latest 5 seconds sending
+ * a new message if it exceeds 20
+ *
+ * In this example, the input stream reads from a topic named "temperature", where the values of messages
+ * represent temperature values; using a 5 seconds "tumbling" window, the maximum value is processed and
+ * sent to a topic named "max" if it exceeds 20
+ *
+ * Before running this example you must create the input topic and the output topic (e.g. via
+ * bin/kafka-topics.sh --create ...), and write some data to the input topic (e.g. via
+ * bin/kafka-console-producer.sh). Otherwise you won't see any data arriving in the output topic.
+ */
+public class TemperatureDemo {
+
+    public static void main(String[] args) throws Exception {
+
+        Properties props = new Properties();
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-temperature");
+        props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+        props.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        props.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0);
+
+        KStreamBuilder builder = new KStreamBuilder();
+
+        KStream<String, String> source = builder.stream("temperature");
+
+        KStream<Windowed<String>, String> max = source
+                .map(new KeyValueMapper<String, String, KeyValue<String, String>>() {
+                    @Override
+                    public KeyValue<String, String> apply(String key, String value) {
+                        return new KeyValue<>("temp", value);
+                    }
+                })
+                .groupByKey()
+                .reduce(new Reducer<String>() {
+                    @Override
+                    public String apply(String value1, String value2) {
+                        if (Integer.parseInt(value1) > Integer.parseInt(value2))
+                            return value1;
+                        else
+                            return value2;
+                    }
+                }, TimeWindows.of(5000))
+                .toStream()
```

Comment:
Just to clarify: I thought `filter` operator is needed for your logic, but `toStream` seems not needed. Since `KTable` itself also supports the `filter` operator so you do not need to force transforming it into a `KStream` before applying `filter`. But your observation is right, and that is by-design of the `KTable#filter` APIs:

https://issues.apache.org/jira/browse/KAFKA-3902

Let me know if you think this does not make sense.